pkg/ui/react-app/src/pages/graph/Panel.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { Component } from 'react';
+
+import { Alert, Button, Col, Nav, NavItem, NavLink, Row, TabContent, TabPane } from 'reactstrap';
+
+import moment from 'moment-timezone';
+
+import ExpressionInput from './ExpressionInput';
+import GraphControls from './GraphControls';
+import { GraphTabContent } from './GraphTabContent';
+import DataTable from './DataTable';
+import TimeInput from './TimeInput';
+import QueryStatsView, { QueryStats } from './QueryStatsView';
+import PathPrefixProps from '../../types/PathPrefixProps';
+import { QueryParams } from '../../types/types';
+
+interface PanelProps {
+  options: PanelOptions;
+  onOptionsChanged: (opts: PanelOptions) => void;
+  useLocalTime: boolean;
+  useDeduplication: boolean;
+  usePartialResponse: boolean;
+  pastQueries: string[];
+  metricNames: string[];
+  removePanel: () => void;
+  onExecuteQuery: (query: string) => void;
+}
+
+interface PanelState {
+  data: any; // TODO: Type data.
```

Comment:
can we type this explicitly?

Reply:
This has been moved as-is from https://github.com/prometheus/prometheus/blob/d996ba20ec9c7f1808823a047ed9d5ce96be3d8f/web/ui/react-app/src/pages/graph/Panel.tsx#L27 which I think is a good thing since it makes this PR easier to review. Could we make a small PR upstream to fix it there @prmsrswt? I think this PR can go forward either way with this.